plutus-contract/src/Wallet/Emulator/ChainIndex/Index.hs

```diff
@@ -102,12 +102,21 @@ insert am item (ChainIndex ci) =
     in ChainIndex (foldl' (\ci' addr -> Map.alter alt addr ci') ci keys)
 
 -- | All transactions that modify the address, from a given slot onwards
-transactionsAt :: ChainIndex -> Slot -> Address -> [ChainIndexItem]
-transactionsAt (ChainIndex mp) sl addr =
-    toList
-    $ unAddressIndex
-    $ fst
-    $ split sl
-    $ snd
-    $ split (pred sl)
-    $ Map.findWithDefault mempty addr mp
+transactionsAt :: ChainIndex -> SlotRange -> Address -> [ChainIndexItem]
+transactionsAt (ChainIndex mp) sl addr = let
+    allItems = Map.findWithDefault mempty addr mp
+    result = case sl of
+        Interval (LowerBound (Finite s1) in1) (UpperBound (Finite s2) in2) ->
+            let low = if in1 then s1 else pred s1
```

Comment:
should this not be `else succ s1` ?

Reply:
Please, take another look. I've fixed this a bit.